zebra-utils/src/bin/scanning-results-reader/main.rs

```diff
@@ -0,0 +1,128 @@
+            UnifiedFullViewingKey::new(Some(dfvk), None).expect("`dfvk` should be `Some`"),
+        )]);
+
+        for (height, txids) in storage.sapling_results(key) {
+            let height = BlockHeight::from_u32(height.0);
+
+            for txid in txids.iter() {
+                let tx = Transaction::read(
+                    &hex::decode(&get_tx_via_rpc(hex::encode(txid.bytes_in_display_order())))
+                        .expect("RPC response should be decodable from hex string to bytes")[..],
+                    BranchId::for_height(&network, height),
+                )
+                .expect("TX fetched via RPC should be deserializable from raw bytes");
+
+                for output in decrypt_transaction(&network, height, &tx, &ufvk_with_acc_id) {
+                    let memo = memo_bytes_to_string(output.memo.as_array());
+
+                    if !memo.is_empty()
+                        // Filter out some uninteresting and repeating memos from ZECPages.
+                        && !memo.contains("LIKE:")
+                        && !memo.contains("VOTE:")
+                        && memo != prev_memo
+                    {
+                        println!("{memo}\n");
+                        prev_memo = memo;
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Trims trailing zeroes from a memo, and returns the memo as a string.
+fn memo_bytes_to_string(memo: &[u8; 512]) -> String {
+    match memo.iter().rposition(|&byte| byte != 0) {
+        Some(i) => std::str::from_utf8(&memo[..=i]).unwrap_or("").to_owned(),
+        None => "".to_owned(),
+    }
+}
+
+/// Uses the `getrawtransaction` RPC to retrieve a transaction by its TXID.
+fn get_tx_via_rpc(txid: String) -> String {
+    // Wrap the TXID with `"` so that [`RawValue::from_string`] eats it.
+    let txid = format!("\"{}\"", txid);
+    let transport = SimpleHttpTransport::builder()
+        .url("127.0.0.1:8232")
+        .expect("URL should be valid")
+        .build();
+    let client = Client::with_transport(transport);
+    let params = [RawValue::from_string(txid).expect("Provided TXID should be a valid JSON")];
+    let request = client.build_request("getrawtransaction", &params);
+    let response = client
+        .send_request(request)
+        .expect("Sending the `getrawtransaction` request should succeed");
+
+    response
+        .result()
+        .expect("Zebra's RPC response should contain a valid result")
+}
+
+/// Converts [`zcash_primitives::consensus::Network`] to [`zebra_chain::parameters::Network`].
+fn zebra_network(
+    network: &zcash_primitives::consensus::Network,
+) -> zebra_chain::parameters::Network {
+    match network {
+        zcash_primitives::consensus::Network::MainNetwork => {
+            zebra_chain::parameters::Network::Mainnet
+        }
+        zcash_primitives::consensus::Network::TestNetwork => {
+            zebra_chain::parameters::Network::Testnet
+        }
+    }
+}
```

Comment:
Optional: This seems like it would be useful in `zebra_chain`, if we don't have this `From` impl there already.

Reply:
You're right. We even have `impl From<Network> for zcash_primitives::consensus::Network`
 in Zebra already. I couldn't use it when the reader was in its own crate. I forgot about that when I was moving the code here.

Reply:
Done in [c0c2101](https://github.com/ZcashFoundation/zebra/pull/8104/commits/c0c21016f13f91b63060152cdaabd982c5901354).